pkgs/servers/vpp/default.nix

```
@@ -0,0 +1,152 @@
+{ lib
+, llvmPackages_14
+, fetchgit
+, fetchurl
+, callPackage
+, cmake
+, dpdk
+, intel-ipsec-mb
+, xdp-tools
+, libbpf
+, libmnl
+, elfutils
+, zlib
+, rdma-core
+, python310
+, python310Packages
+, openssl
+, libuuid
+, subunit
+, pkg-config
+, libpcap
+, jansson
+, libnl
+, libdaq
+, srtp
+, mbedtls_2
+, check
+, nixosTests
+}:
+
+let
+  stdenv = llvmPackages_14.stdenv;
+
+  xdp-tools' = xdp-tools.overrideAttrs (old: rec {
+    postInstall = ''
+      # Drop unfortunate references to glibc.dev/include at least from $lib
+      nuke-refs "$lib"/lib/bpf/*.o
+    '';
+  });
+
+  intel-ipsec-mb' = intel-ipsec-mb.overrideAttrs (old: rec {
+    makeFlags = old.makeFlags ++ [
+      "SHARED=n"
+    ];
+  });
+
+  dpdk' = dpdk.overrideAttrs (old: rec {
+    version = "22.07";
+    src = fetchurl {
+      url = "https://fast.dpdk.org/rel/dpdk-${version}.tar.xz";
+      sha256 = "sha256-n2Tf3gdf21cIy2Leg4uP+4kVdf7R4dKusma6yj38m+o";
+    };
+  });
+
+  rdma-core' = rdma-core.overrideAttrs (old: rec {
+    cmakeFlags = old.cmakeFlags ++ [
+      "-DENABLE_STATIC=1"
```

No need to build the shared libraries.
```suggestion
      "-DENABLE_STATIC=1"
      "-DBUILD_SHARED_LIBS:BOOL=false"
```